tgui/packages/tgui/interfaces/Sleeper.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author BenLubar (https://github.com/BenLubar)
+ * @license ISC
+ */
+
+import { useBackend } from "../backend";
+import { Box, Button, Section, Knob, ProgressBar, LabeledList, Flex, Icon, TimeDisplay, HealthStat } from "../components";
+import { Window } from "../layouts";
+import { formatTime } from "../format";
+
+const damageNum = num => num <= 0 ? "0" : num.toFixed(1);
+
+export const Sleeper = (props, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    sleeperGone,
+    hasOccupant,
+    occupantStat,
+    health,
+    oxyDamage,
+    toxDamage,
+    burnDamage,
+    bruteDamage,
+    recharging,
+    rejuvinators,
+    isTiming,
+    time,
+    timeStarted,
+    timeNow,
+  } = data;
+
+  const curTime = Math.max(timeStarted
+    ? (time || 0) + timeStarted - timeNow
+    : (time || 0), 0);
+  const canInject = hasOccupant && !isTiming && !recharging && occupantStat < 2;
+
+  return (
+    <Window
+      theme="ntos"
+      width={440}
+      height={440}>
+      <Window.Content>
+        <Section title="Occupant Statistics"
+          buttons={
+            <Button
+              icon="eject"
+              align="center"
+              color="good"
+              disabled={!hasOccupant || !!isTiming}
+              onClick={() => act("eject")}>
+              {"Eject"}
+            </Button>
+          }>
+          {!hasOccupant && (sleeperGone ? "Check connection to sleeper pod." : "The sleeper is unoccupied.")}
+          {!!hasOccupant && (
```

Comment:
@stylemistake What's TG's take on use of `!!` on boolean things taken from `ui_data`?

Reply:
must use it